clippy_lints/src/copies.rs

```diff
@@ -360,3 +364,36 @@ where
 
     None
 }
+
+fn search_same_list<T, Hash, Eq>(exprs: &[T], hash: Hash, eq: Eq) -> Option<Vec<(&T, &T)>>
+where
+    Hash: Fn(&T) -> u64,
+    Eq: Fn(&T, &T) -> bool,
+{
+    let mut match_expr_list: Vec<(&T, &T)> = Vec::new();
+
```

Comment:
Compared to the `search_same` implementation you dropped the 'common cases' check - is this intentional?

Reply:
That's why I miss the first error case on the list. I'm trying to make a function out of it in order to not duplicate this code and use it in both methods.